### DISCUSSION TITLE
[@types/bootstrap] jQuery Global `$` conflicts with Vue.js reactivity transform

### DISCUSSION THREAD

**Discussion by tgrelka** [DefinitelyTyped/DefinitelyTyped#60443] [Issues with a @types package] [answered]
I’m using Bootstrap 5 together with Vue.js in a project.
To integrate the JS modules, I’m using `@types/bootstrap` package for TS definitions.
Additionally, I’m using the [Vue.js Reactivity Transform](https://vuejs.org/guide/extras/reactivity-transform.html) feature, which adds some compile-time macros, on of which has the signature `$(...)`.

However, ever since installing `@types/bootstrap`, I’m getting Typescript errors because apparently, `$()` is now recognised as `JQueryStatic`, because the Bootstrap types package references these.
However, I do not have jQuery installed, and bootstrap will only enable its jQuery plugins if it detect
s an installation, so I don’t think `@types/bootstrap` should register any globals for `$()` or other jQuery functions.

![CleanShot 2022-05-20 at 12 19 49](https://user-images.githubusercontent.com/22134144/169507923-4fb0a5f3-51aa-4a71-b7bf-efd3e75fba65.png)

I tried redefining `$` in an `env.d.ts`, but it didn’t work and I’m still stuck with the jQuery global. So additionally, any quick mitigation I can do on the user side would also be greatly appreciated.

[a comment by an automated account is not shown]

**Comment by martin-badin** [comment]
I think we should remove the JQuery reference and it should be an optional reference

**Reply by martin-badin** [reply]
I created PR to remove the reference 
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60444

**Reply by tgrelka** [reply]
That seems like a good choice to me, seeing how one will likely have the jQery types installed anyways if they’re using it together with Bootstrap in TS.
So the reference wasn’t needed for anything related to getting Bootstrap’s jQuery plugin types to work?

**Comment by peterblazejewicz** [comment, marked as answer]
fixed in #60444
